lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -2108,29 +2112,31 @@ public function createEntity($className, array $data, &$hints = [])
                 isset($hints[Query::HINT_REFRESH])
                 && isset($hints[Query::HINT_REFRESH_ENTITY])
                 && ($unmanagedProxy = $hints[Query::HINT_REFRESH_ENTITY]) !== $entity
-                && $unmanagedProxy instanceof Proxy
+                && $unmanagedProxy instanceof GhostObjectInterface
                 && $this->isIdentifierEquals($unmanagedProxy, $entity)
             ) {
+                // @TODO messy, but for now we try continuing operation as normal
+                // @TODO ideally, all this scenario should be completely gone
+                $entity = $unmanagedProxy;
                 // DDC-1238 - we have a managed instance, but it isn't the provided one.
                 // Therefore we clear its identifier. Also, we must re-fetch metadata since the
                 // refreshed object may be anything
-                foreach ($class->identifier as $fieldName) {
-                    $property = $class->getProperty($fieldName);
+                // @TODO DDC-1238 seems to be invalid.
+                // @TODO We shouldn't touch the object state, but instead just consider it "unmanaged" and skip further hydration
+//                foreach ($class->identifier as $fieldName) {
```

Comment:
This block was super-broken: we just want an un-managed instance, but we really don't want to change its identifier. Replacing the identifier = bad.